File: c++/leetcode/771. Jewels and Stones.cpp
class Solution
{
public:
  int numJewelsInStones(string J, string S)
  {
    int count = 0;
    set<char> st(begin(J), end(J));

    for (auto w : S)
    {
      if (st.count(w))
        count++;
    }

    return count;
  }
};

File: go/leetcode/771. Jewels and Stones.go
func numJewelsInStones(J string, S string) int {
    count:=0
    for _, val := range S {
        for _, val1 := range J { 
            if val == val1 {
                count++
            }
        }
    }
    return count
}


///
import "strings"

func numJewelsInStones(J string, S string) int {
    count := 0
    
    for i := 0; i < len(S); i++ {
        if(strings.Contains(J, string(S[i]))) {count++}
    }
    
    return count
}